mailchimp/mailchimp-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/mailChimp/MailChimpAuthenticator.java

```diff
@@ -0,0 +1,306 @@
+
+        Property callbackUrl = new Property();
+        callbackUrl.setDisplayName("Callback URL");
+        callbackUrl.setName(IdentityApplicationConstants.OAuth2.CALLBACK_URL);
+        callbackUrl.setDescription("Enter the callback url");
+        callbackUrl.setDisplayOrder(2);
+        configProperties.add(callbackUrl);
+
+        return configProperties;
+    }
+
+    @Override
+    protected void processAuthenticationResponse(HttpServletRequest request, HttpServletResponse response,
+                                                 AuthenticationContext context) throws AuthenticationFailedException {
+        try {
+            Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
+            String clientId = authenticatorProperties.get(OIDCAuthenticatorConstants.CLIENT_ID);
+            String clientSecret = authenticatorProperties.get(OIDCAuthenticatorConstants.CLIENT_SECRET);
+            String tokenEndPoint = getTokenEndpoint(authenticatorProperties);
+            String callbackUrl = getCallbackUrl(authenticatorProperties);
+            OAuthAuthzResponse authorizationResponse = OAuthAuthzResponse.oauthCodeAuthzResponse(request);
+            String code = authorizationResponse.getCode();
+            OAuthClientRequest accessRequest =
+                    getAccessRequest(tokenEndPoint, clientId, code, clientSecret, callbackUrl);
+            OAuthClient oAuthClient = new OAuthClient(new URLConnectionClient());
+            OAuthClientResponse oAuthResponse = getOauthResponse(oAuthClient, accessRequest);
+            String accessToken = oAuthResponse.getParam(OIDCAuthenticatorConstants.ACCESS_TOKEN);
+            if (StringUtils.isBlank(accessToken)) {
+                throw new AuthenticationFailedException("Access token is empty or null");
+            }
+            context.setProperty(OIDCAuthenticatorConstants.ACCESS_TOKEN, accessToken);
+            Map<ClaimMapping, String> claims = getSubjectAttributes(oAuthResponse, authenticatorProperties);
+            String email = claims.get(ClaimMapping.build(MailChimpAuthenticatorConstants.MailChimp_EMAIL, MailChimpAuthenticatorConstants.MailChimp_EMAIL, (String) null, false));
+            AuthenticatedUser authenticatedUserObj =
+                    AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier(email);
+            authenticatedUserObj.setAuthenticatedSubjectIdentifier(email);
+            authenticatedUserObj.setUserAttributes(claims);
+            context.setSubject(authenticatedUserObj);
+        } catch (OAuthProblemException e) {
+            throw new AuthenticationFailedException("Authentication process failed", e);
+        }
+    }
+
+    @Override
+    protected Map<ClaimMapping, String> getSubjectAttributes(OAuthClientResponse token, Map<String, String> authenticatorProperties) {
+        HashMap claims = new HashMap();
+
+        try {
+            String e = token.getParam("access_token");
```

Comment:
Fixed in https://github.com/wso2-attic/is-connectors/pull/69/commits/c9a7dd98c0c10e4efde3f0260cff94bb966477a4